modules/nf-core/picard/collectinsertsizemetrics/main.nf

```diff
@@ -0,0 +1,61 @@
+process PICARD_COLLECTINSERTSIZEMETRICS {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::picard=3.0.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/picard:3.0.0--hdfd78af_1' :
+        'quay.io/biocontainers/picard:3.0.0--hdfd78af_1' }"
+
+    input:
+    tuple val(meta), path(bam)
+
+    output:
+    tuple val(meta), path("*.txt"), emit: metrics
+    tuple val(meta), path("*.pdf"), emit: histogram
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def avail_mem = 3072
+    if (!task.memory) {
+        log.info '[Picard CollectInsertSizeMetrics] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
+    } else {
+        avail_mem = (task.memory.mega*0.8).intValue()
+    }
+    """
+    picard \\
+        -Xmx${avail_mem}M \\
+        CollectInsertSizeMetrics \\
+        $args \\
+        --INPUT $bam \\
+        --OUTPUT ${prefix}.insert_size_metrics.txt \\
+        --Histogram_FILE ${prefix}.insert_size_histogram.pdf \\
```

Comment:
```suggestion
        --OUTPUT ${prefix}.txt \\
        --Histogram_FILE ${prefix}.pdf \\
```

You should leave this up to the pipeline developer to add themselves via the prefix